### PR TITLE
Token order and Token Source updates

### PR DESCRIPTION
**Pull request by njones**
Fix the handling of token source ordering so that it can be determined by the user of the library
Fix so that token sources can be removed
Fix so that a custom token name can be used and override the default names
Refactor error propagation so users of the library can see what is causing an error
Add a test to confirm the new code works as expected

Resolves: #13

### CODE REVIEW

**Comment by njones** [comment]
I just realized that you asked for a README update as well... I will push something out later tonight for that.

File: validator_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+func TestAuthorize(t *testing.T) {
```
Comment (greenpau) [inline validator_test.go:14]: @njones . 👍

File: validator.go [hunk 2]
```diff
@@ -16,41 +17,65 @@ const (
...
-	v.AuthorizationHeaders = make(map[string]bool)
-	v.Cookies = make(map[string]bool)
-	v.QueryParameters = make(map[string]bool)
-	v.AuthorizationHeaders["access_token"] = true
-	v.AuthorizationHeaders["jwt_access_token"] = true
-	v.Cookies["access_token"] = true
-	v.Cookies["jwt_access_token"] = true
-	v.QueryParameters["access_token"] = true
-	v.QueryParameters["jwt_access_token"] = true
+func NewTokenValidator(m *AuthProvider) *TokenValidator {
```
Comment (greenpau) [inline validator.go:60]: @njones , as I mentioned earlier, please remove the arguments from this function.

**Comment by njones** [comment]
> @njones , let's not add the `AuthProvider`. I want this function without arguments.
> 
> It is best to have pointer methods later. I would prefer `func (v *TokenValidator) ConfigureWhatever`.

Does `func (v *TokenValidator) SetTokenName(name string) error` work? Which is using a "Setter" function to setup everything that needs to happen with setting a token name?

**Comment by greenpau** [comment]
> Does func (v *TokenValidator) SetTokenName(name string) error work? Which is using a "Setter" function to setup everything that needs to happen with setting a token name?

@njones , yes, "Set*" works.

**Comment by greenpau** [comment]
@njones , thank you for the contribution! ;-)

**Pull request merged**